src/libstd/sys/unix/args.rs

```diff
@@ -46,8 +46,63 @@ impl DoubleEndedIterator for Args {
 }
 
 #[cfg(any(target_os = "linux",
-          target_os = "android",
-          target_os = "freebsd",
+          target_os = "android"))]
+mod imp {
+    use crate::os::unix::prelude::*;
+    use crate::ptr;
+    use crate::ffi::{CStr, OsString};
+    use crate::fs;
+    use crate::marker::PhantomData;
+    use super::Args;
+
+    use crate::sys_common::mutex::Mutex;
+
+    static mut ARGC: isize = 0;
+    static mut ARGV: *const *const u8 = ptr::null();
+    // We never call `ENV_LOCK.init()`, so it is UB to attempt to
```

Comment:
*LOCK

Reply:
I just copied the existing code for non-macOS Unix.